jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2472,7 +2475,8 @@ ecma_builtin_array_prototype_object_reduce (ecma_value_t this_arg, /**< this arg
         bool k_present = false;
 
         /* 8.b */
-        while (!k_present && index < len && ecma_is_value_empty (ret_value))
+        bool has_items_left = start_from_left ? (index < len) : (index != UINT32_MAX);
```

Comment:
has_items_left ? I also don't like the `UINT32_MAX` comparison, sometimes the compilers decides that certain overflowed values cannot happen.

Reply:
Would it be possible to use an indexing from right? So go from 0 - len, and substract the current value from the len? I would not use int64 because it is not a native type on 32 bit systems.

Reply:
Actually the whole `has_items_left` could be replaced with a counter -- I'll push an update.